src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int salary = 0;
+        String[] info;
+        StringBuilder employeeSalary = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo);
+
+        for (String name: names) {
+            for (String dataPerDay: data) {
+                info = dataPerDay.split(" ");
+
+                if (name.equals(info[1])
+                        && dateConvert(info[0]).compareTo(dateConvert(dateFrom)) >= 0
+                        && dateConvert(info[0]).compareTo(dateConvert(dateTo)) <= 0) {
+                    salary += Integer.parseInt(info[2]) * Integer.parseInt(info[3]);
```

Comment:
add constants for indexes
also you can parse dateFrom and dateTo once - in the beginning of the method, it wont change
parse date from info before if as well, no need to do 2 operations that give the same result